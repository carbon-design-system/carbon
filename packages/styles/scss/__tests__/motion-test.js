/**
 * Copyright IBM Corp. 2018, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @jest-environment node
 */

'use strict';

const { SassRenderer } = require('@carbon/test-utils/scss');

const { render } = SassRenderer.create(__dirname);

describe('@carbon/styles/scss/motion', () => {
  test('Public API', async () => {
    const { get } = await render(`
      @use 'sass:meta';
      @use '../motion';

      $_: get('api', (
        variables: meta.module-variables('motion'),
        functions: (
          motion: meta.function-exists('motion', 'motion'),
        ),
        mixins: (
          motion: meta.mixin-exists('motion', 'motion'),
        ),
      ));
    `);

    const { value: api } = get('api');

    expect(api).toMatchSnapshot();
  });
});
