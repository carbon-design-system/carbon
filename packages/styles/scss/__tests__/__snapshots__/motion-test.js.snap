// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@carbon/styles/scss/motion Public API 1`] = `
Object {
  "functions": Object {
    "motion": true,
  },
  "mixins": Object {
    "motion": true,
  },
  "variables": Object {
    "duration-fast-01": "70ms",
    "duration-fast-02": "110ms",
    "duration-moderate-01": "150ms",
    "duration-moderate-02": "240ms",
    "duration-slow-01": "400ms",
    "duration-slow-02": "700ms",
    "ease-in": "cubic-bezier(0.25, 0, 1, 1)",
    "ease-out": "cubic-bezier(0, 0, 0.25, 1)",
    "easings": Object {
      "entrance": Object {
        "expressive": "cubic-bezier(0, 0, 0.3, 1)",
        "productive": "cubic-bezier(0, 0, 0.38, 0.9)",
      },
      "exit": Object {
        "expressive": "cubic-bezier(0.4, 0.14, 1, 1)",
        "productive": "cubic-bezier(0.2, 0, 1, 0.9)",
      },
      "standard": Object {
        "expressive": "cubic-bezier(0.4, 0.14, 0.3, 1)",
        "productive": "cubic-bezier(0.2, 0, 0.38, 0.9)",
      },
    },
    "standard-easing": "cubic-bezier(0.5, 0, 0.1, 1)",
    "transition-base": "250ms",
    "transition-expansion": "300ms",
  },
}
`;
